src/ucp/core/ucp_ep.c

```diff
@@ -3404,3 +3404,17 @@ ucs_status_t ucp_ep_query(ucp_ep_h ep, ucp_ep_attr_t *attr)
 
     return UCS_OK;
 }
+
+unsigned ucp_ep_rkey_pack_uct_flags(ucp_ep_h ep)
+{
+    unsigned flags = 0;
+
+    if ((ucp_ep_config(ep)->key.err_mode ==
+         UCP_ERR_HANDLING_MODE_PEER) &&
+        /* and at least 1 lane is connect2iface */
+        (ucp_ep_config(ep)->p2p_lanes == 0)) {
+        flags |= UCT_MD_MKEY_PACK_FLAG_INVALIDATE;
```

Comment:
can we cache these flags in ep_config
`ucp_ep_config(ep)->uct_rkey_pack_flags`

then, remove this wrapper function